### DISCUSSION TITLE
[Middleware] Guests only routes?

### DISCUSSION THREAD

**Discussion by goustaff** [nextauthjs/next-auth#10628] [Help]
Hello!

The Next Auth documentation says:

https://next-auth.js.org/configuration/nextjs#basic-usage
> "[...] Now you will still be able to visit every page, but only `/dashboard` will require authentication."

Okay, perfect.
But if I want to do the opposite, with a page like `/landing-page/dashboard` which should be only visible for guests (i.e: unauthenticated users), and redirecting all logged in users to `/dashboard`, what should I do?

I would like a middleware approach, to preserve SSG.
Is it supported by Next Auth?

---

EDIT: Hmmm, maybe it could be easily implemented with Next Auth v5, using `req.auth`
```ts
import { auth } from "./auth"
 
export default auth((req) => {
  // req.auth
})
 
// Optionally, don't invoke Middleware on some paths
export const config = {
  matcher: ["/((?!api|_next/static|_next/image|favicon.ico).*)"],
}
```

I'll try to figure out what I should do with Next Auth v4. :x

**Comment by JasperAlexander** [comment]
You can also use the middleware config for v4.
```ts
export { default } from "next-auth/middleware"

export const config = {
  matcher: ["/((?!api|_next/static|_next/image|favicon.ico).*)"],
}
```

**Reply by goustaff** [reply]
That's fine when you have only one middleware. But whenever you'll extend the matcher, auth will be impacted too…

I use i18n, which means that I can't use a narrowed matcher like so.

It is also not really solving the issue here. The matter is to call auth middleware on both guests only and auth only routes, and trigger different behaviors.

**Reply by goustaff** [reply]
https://github.com/Tirraa/dashboard_rtm/blob/main/src/middlewares/withProtectedRoutes.ts
